### ISSUE TITLE
[Idea] MUL-T's Power Mode, and it's place in MUL-T's kit.

### ISSUE THREAD

**Issue by Coffe-e** [Moffein/RiskyMod#38]
Personally, I've always felt that MUL-T's Power Mode skill always was meant to be a Utility instead of a Special, as a dichotomy of Transport Mode's "Speed and Defense at the cost of Offense" relative to Power Mode's "Defense and Offense at the cost of Speed" made more sense than it replacing his special. In addition, it takes away MUL-T's core uniqueness of his kit: Being able to use two Primaries/equipment, and swapping them based on the occasion. It takes away his enhanced choice of options in each scenario, instead of just trading mobility for DPS, and I think that's a bit sad.
Ofc this prolly is mostly my personal opinion, but I haven't seen many people, if any at all even attempt to try it out as a utility, and i'm a bit sad that I haven't been able to experiment, and potentially see why it's not a good idea myself, or if it'd fit perfectly. Likely would end up being a config: default off type deal, but I think it'd still be welcome. I'd love to hear your thoughts on Power Mode and it's place though!

**Comment by Moffein** [comment]
I agree with your point about weapon swapping and MUL-T's identity. Power Mode is a fun skill, but it feels slightly tacked-on with how awkwardly it interacts with MUL-T's weapons selection/equipment selection.

I think Power Mode as a Utility would lilkely completely outshine Transport mode due to the constant bonus DPS stance being a lot stronger than the brief dash from Transport, even if Power Mode were to have no armor.

**Comment by Moffein** [comment]
You make some good points. Probably won't be able to try it out since the internal code behind it is too jank to deal with (Power Mode simply lets you use the unused second Primary from Retool, so making the Power Mode primaries swappable would likely involve a lot of hacky code).

Feel free to post future ideas here.

**Comment by Coffe-e** [comment]
Yeah, I figured that'd at least be a little jank in code even if it was an interesting idea. If you find an easier/cleaner way to do it, i'd happily do a bunch of test-runs with MUL-T to see how it goes though! As for the other ideas, I'll post them below. Changing the name of this entry as well.

[Enemy Spawning, Movement speeds, and Downtime]
I'm sure you know how ROR2's spawning can be with enemies, having gaps of like 10-15 seconds without any enemies whatsoever sometimes. Or, having situations where enemies are across the map and you don't really have a quick way to get to them in order to get gold for the stage. I had a few ideas for these two issues, and while mitigated in my use of mods, I still feel that sometimes it can still happen.

First, for gaps in enemies spawning, I think that some sort of non-director measure would be best, as the director using credits to spawn things Is what i believe ends up causing this in the first place. Maybe having a mechanic where, if there is a low number of enemies within a certain distance of the player for around 15 seconds or more, the game will forcibly spawn some fodder enemies (beetles, lemurians, etc) in order to keep the player engaged while the director builds up credits to spawn more interesting enemies, or the enemies in the distance finally catch up.

Second, possibly in addition to the first idea, you could have enemies get a large movement speed buff when further than a certain distance from the player, in order to make that process of waiting for them to get to your location take much shorter. I dunno if when they're that far, they do try to get to you, or if they idle, but I think if they're that far they should prolly auto-target you so they aren't just taking up the enemy cap.

[Player Speed, and Player Characters]
There's a pretty large dichotomy between characters with lots of mobility, and characters with very little. If you aren't mobile, you're defensive. If you are mobile, you're squishy. But with that, I feel there's two characters who have a rougher time because of another factor: **Size**. Pretty much every humanoid character has a very small hitbox, which makes sense, and the two outliers whom I feel are different enough/impactful enough to mention: Acrid, and MUL-T (Maybe REX too but I haven't played them in a while). Due to their size, hitbox, and camera placement, they end up feeling like the slowest characters in the game. (Acrid having it mitigated by leap, and MUL-T by transport mode) But I think think the universal speed values bring a different problem: Without Items, or their movement abilities, these characters get hit _more _often__ and _easier_ than any others. This doesn't help the feeling of sluggishness without the use of movement abilities, and I feel it impacts their gameplay negatively in a noticeable way as a result.
Personally, I think that these characters, because of their increased hitbox size and general reliance on their movement abilities to dodge, should have an increased base movement speed. It doesn't have to be something big, maybe 8-9m/s instead of the universal 7m/s. I feel it would go a long way in making them feel generally nicer to play, and eliminate that aspect of getting hit easier than the others.

**Comment by Moffein** [comment]
[Enemy Spawning]
Circumventing the game's director system is a no-go since it has the potential to cause a lot of incompatibilities. Something I've noticed is that RoR1 has the exact same spawnrate problems that 2 has, but it's much less pronounced since the game encourages you to activate the teleporter as soon as possible, which boosts spawnrates to a good level. Earlier versions of this mod had higher spawnrates by default, but it got toned down because it caused certain survivors to fall behind, being unable to deal with all the enemies that spawned. I think the solution to this problem would be for modders to add more custom interactables/side objectives to maps so that players have something to do other than wait for enemies to spawn.

[Movement Speeds and Downtime]
Movement speeds has the potential to look jank since most enemies have no sprint animation and end up having a really weird fast walk animation when you boost their speed so I'll probably pass on doing that, but I do like the idea in theory. Making enemies always aware of the player sounds good, since there's no reason for Lemurians to randomly idle in an obscure corner of the map. I'll see what I can do about that.

[Player Speed]
This is how it used to be in Early Access, with MUL-T having a higher move speed than other characters, but it got removed. I'll probably lean on keeping it gone, mainly because it's something that's hard to get a feel for balancewise, while the actual benefit is dubious (even when MUL-T had 9 move speed, people were still saying he felt slow and sluggish).

**Comment by Coffe-e** [comment]
All of these reasons are pretty understandable, and I do agree with a lot of it. Though, I'm kinda sad with the movement speed reason being that the animation is jank. you give a lot of feedback on these discussions and I love that, it's pretty fun to discuss with you. I have a couple more things i'd like to say before I prolly leave this for a bit, but it's been a blast talking with ya.

[Map Size, and general traversal]
I'll just say it here: I think map traversal early on suuuuucks, especially for characters without movement abilities. Some sort of out of combat speed buff like red whip, or a sprint speed buff would make earlier stages feel so much better in my eyes. This issue is ESPECIALLY bad for the dlc snow stage. It's a great stage, but man I usually end it at 7 min thanks to the new tele changes. if not a speed buff, maybe something to just generally help with exploration to make it faster? unsure what it'd be besides movespeed though
